GitHubContents.ps1

```diff
@@ -269,37 +275,61 @@ function Set-GitHubContent
         the background, enabling the command prompt to provide status information.
         If not supplied here, the DefaultNoStatus configuration property value will be used.
 
+     .INPUTS
+        GitHub.Branch
+        GitHub.Content
+        GitHub.Event
+        GitHub.Issue
+        GitHub.IssueComment
+        GitHub.Label
+        GitHub.Milestone
+        GitHub.PullRequest
+        GitHub.Project
+        GitHub.ProjectCard
+        GitHub.ProjectColumn
+        GitHub.Release
+        GitHub.Repository
+
+    .OUTPUTS
+        GitHub.Content
+
     .EXAMPLE
-        Set-GitHubContent  -Path README.md -OwnerName microsoft -RepositoryName PowerShellForGitHub -CommitMessage 'Adding README.md' -Content '# README' -BranchName master
+        Set-GitHubContent -OwnerName microsoft -RepositoryName PowerShellForGitHub -Path README.md -CommitMessage 'Adding README.md' -Content '# README' -BranchName master
 
         Sets the contents of the README.md file on the master branch of the PowerShellForGithub repository.
 #>
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute('PSReviewUnusedParameter', '',
+        Justification = 'One or more parameters (like NoStatus) are only referenced by helper
+        methods which get access to it from the stack via Get-Variable -Scope 1.')]
     [CmdletBinding(
         SupportsShouldProcess,
         PositionalBinding = $false,
         DefaultParameterSetName = 'Elements')]
```

Comment:
Looks like you need to add a test involving the pipeline.  If you try to pipe in the repo to this, it still wants `OwnerName`/`RepositoryName` due to this default that you have here.

Reply:
I recommend an additional test for any parameter that can be piped in (that's what I ended up doing with the pipeline change which is why the test count exploded...but it caught a number of bugs along the way).